plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/ArchiveCaseAction.java

```diff
@@ -0,0 +1,247 @@
+        ",", set.stream().map(name -> ("\"" + name + "\"")).collect(Collectors.toList()));
+  }
+
+  @Override
+  public ActionState perform(
+      ActionServices services, Duration lastGeneratedByOlive, boolean isOliveLive) {
+    final var overloaded = services.isOverloaded("all", "nabu");
+    if (!overloaded.isEmpty()) {
+      this.errors =
+          Collections.singletonList("Overloaded services: " + String.join(", ", overloaded));
+      return ActionState.THROTTLED;
+    }
+
+    HttpRequest.BodyPublisher body;
+    try {
+      body = HttpRequest.BodyPublishers.ofString(createRequestBody());
+    } catch (final Exception e) {
+      e.printStackTrace();
+      this.errors = Collections.singletonList(e.getMessage());
+      return ActionState.FAILED;
+    }
+    final var baseUrl = owner.get().NabuUrl();
+
+    final var request =
+        HttpRequest.newBuilder(URI.create(baseUrl + "/case"))
+            .header("Content-type", "application/json")
+            .header("Accept", "application/json")
+            .POST(body)
+            .build();
+
+    owner.log("NABU REQUEST: " + request, LogLevel.DEBUG, null);
+    try (var timer = NabuRequestTime.start(baseUrl)) {
+      var response =
+          HTTP_CLIENT.send(request, new JsonListBodyHandler<>(MAPPER, NabuCaseArchiveDto.class));
+      if (response.statusCode() == 409) {
+        return ActionState.HALP;
+      } else if (response.statusCode() / 100 != 2) {
+        NabuRequestErrors.labels(baseUrl).inc();
+        showHTTPError(response, baseUrl);
+        return ActionState.FAILED;
+      } else if (response.statusCode() == 201) {
+        return ActionState.INFLIGHT;
+      } else if (response.statusCode() == 200) {
+        final var results = response.body().get().collect(Collectors.toList());
+        return actionStatusFromArchive(results.get(0));
+      } else {
+        return ActionState.FAILED;
+      }
+    } catch (final Exception e) {
+      e.printStackTrace();
+      this.errors = Collections.singletonList(e.getMessage());
+      NabuRequestErrors.labels(baseUrl).inc();
+      return ActionState.FAILED;
+    }
+  }
+
+  @Override
+  public int priority() {
+    return 0;
+  }
+
+  @Override
+  public long retryMinutes() {
+    return 10;
+  }
+
+  private void showHTTPError(HttpResponse<?> response, String url)
+      throws UnsupportedOperationException, IOException {
+    final List<String> errors = new ArrayList<>();
+    final Map<String, String> labels = new TreeMap<>();
+    labels.put("url", url);
+    owner.log("HTTP error: " + response.statusCode(), LogLevel.ERROR, labels);
+    errors.add("HTTP error: " + response.statusCode());
+    this.errors = errors;
+  }
+
+  @Override
+  public ObjectNode toJson(ObjectMapper mapper) {
+    final var node = mapper.createObjectNode();
+    node.put("type", "nabu-archive");
+    node.put("caseId", caseId);
+    node.put("requisitionId", requisitionId);
+    limsIds.forEach(node.putArray("limsIds")::add);
```

Comment:
putArray says it replaces an old value if it exists (https://fasterxml.github.io/jackson-core/javadoc/1.9/org/codehaus/jackson/node/ObjectNode.html#putArray(java.lang.String), confirming this writes out all the values and doesn't keep overwriting as part of the forEach?

Reply:
Yep they all print out like:
```
"workflowRunIdsForOffsiteArchive": [
    "vidarr:clinical/run/bacc8336b59960f18e9f7eef4bd9f59a429ec6986575693252d463e6b338ec3e",
    "vidarr:clinical/run/ef6b58086b6723e7c92df3c6c10ea584b999d04118141df6404b1db5679b83ec"
  ],
```